extensions/podman/packages/extension/src/podman-configuration.ts

```diff
@@ -105,6 +107,16 @@ export class PodmanConfiguration {
     }
   }
 
+  async updateProxySettigsSync(proxy: ProxySettings | undefined): Promise<void> {
+    if (!this.updatedProxySettingsPromise) {
+      this.updatedProxySettingsPromise = this.updateProxySettings(proxy);
```

Comment:
We are not awaiting for `this.updatedProxySettingsPromise`, this is a problem? like do we want the `updateProxySettigsSync` to finish before the updateProxySettings promise resolved ?

Reply:
I feel like this is an issue to have several call, I would argue we should ignore the second one, or throw an error if the resource is locked.

The root cause is probably the fact that it is called twice.

Otherwise if this is not feasable, we should improve the method signature.

```
queueProxySettingsUpdate(proxy: ProxySettings | undefined): void;
```

This is not a sync operation, but a queue, as we chain the promises until resolve.
Please also add a cleaning mechanism, the promise should be undefined if all are resolved